draco/learn.py

```diff
@@ -0,0 +1,231 @@
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib.colors import ListedColormap
+from sklearn import svm
+from sklearn.decomposition import PCA
+from sklearn.model_selection import train_test_split
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class DracoLearn:
+    def __init__(
+        self,
+        feature_vecs: pd.DataFrame,
+    ):
+        """
+        Initializes a new debugging instance.
+
+        :param feature_vecs: a ``Dataframe`` of pairs of training data.
+            the format should be as the output of ``data_utils.pairs_to_vec``
+        """
+        self.feature_vecs = feature_vecs
+
+    def train_model(self, X, test_size: float = 0.3, C: float = 1, quiet=False):
+        """Given features X and labels y, train a linear model to classify them
+        Args:
+            X: a N x M matrix, representing feature vectors
+            y: a N vector, representing labels
+            test_size: the fraction of test data
+        """
+
+        X_train, X_dev = train_test_split(X, test_size=test_size, random_state=1)
+
+        size = len(X_train)
+
+        y_train = np.ones(size)
+
+        # flip a few examples at random
+        idx = np.ones(size, dtype=bool)
+        idx[: int(size / 2)] = False
+        np.random.shuffle(idx)
+
+        X_train[idx] = -X_train[idx]
+        y_train[idx] = -y_train[idx]
+
+        clf = svm.LinearSVC(C=C, fit_intercept=False)
+        clf.fit(X_train, y_train)
+
+        if not quiet:
+            print("Train score: ", clf.score(X_train, y_train))
+            if test_size > 0:
+                print("Dev score: ", clf.score(X_dev, np.ones(len(X_dev))))
+
+        return clf
+
+    def train_and_plot(self, test_size: float = 0.3):
```

Comment:
We should separate the training from the plotting. This way people can use only the training methods which is what most people will want. Moreover, we can then test only the training method. The plotting could either live in a notebook (I think that makes sense given that it's not polished yet) or I a separate method that we mark as not tested.